### PR TITLE
adding within example

### PR DESCRIPTION
**Pull request by tsuz**
A PR for adding a `within` expression example to show POIs along a route

Tackles this: https://github.com/mapbox/ios-sdk-examples/issues/393

Screen Shot 2020-08-28 at 23 03 07

### CODE REVIEW

**Comment by captainbarbosa** [comment]
Moved to [Maps SDK live board](https://github.com/orgs/mapbox/projects/519) for prioritization.

File: Examples/Swift/POIAlongRouteExample.swift [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        camera.centerCoordinate.longitude = -122.6418837958432
+        self.mapView.setCamera(camera, animated: false)
+        self.mapView.setZoomLevel(15.062187320447523, animated: false)
+    }
+
+    func drawPolyline(geoJson: Data) {
+        // Add our GeoJSON data to the map as an MGLGeoJSONSource.
+        // We can then reference this data from an MGLStyleLayer.
+
+        // MGLMapView.style is optional, so you must guard against it not being set.
```
Comment (pjleonard37) [inline Examples/Swift/POIAlongRouteExample.swift:68]: Helpful comment 👍

File: Examples/Swift/POIAlongRouteExample.swift [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        guard let poiLayer = self.mapView.style?.layer(withIdentifier: "poi-label") as? MGLSymbolStyleLayer else {
+            return
+        }
+        style.insertLayer(layer, below: poiLayer)
+        style.insertLayer(dashedLayer, above: layer)
+        style.insertLayer(casingLayer, below: layer)
+
+    }
+
+    func restrictPOIVisibleShape() {
```
Comment (pjleonard37) [inline Examples/Swift/POIAlongRouteExample.swift:123]: Perhaps add a comment explaining the steps in this function

File: Examples/ObjectiveC/POIAlongRouteExample.m [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    // Dash pattern in the format [dash, gap, dash, gap, ...]. You’ll want to adjust these values based on the line cap style.
+    dashedLayer.lineDashPattern = [NSExpression expressionForConstantValue:@[@0, @1.5]];
+    
+    MGLSymbolStyleLayer *poiLayer = (MGLSymbolStyleLayer *)[self.mapView.style layerWithIdentifier:@"poi-label"];
+    [self.mapView.style insertLayer:layer belowLayer:poiLayer];
+    [self.mapView.style insertLayer:dashedLayer belowLayer:layer];
+    [self.mapView.style insertLayer:casingLayer belowLayer:layer];
+}
+
+- (void)restrictPOIVisibleShape {
```
Comment (pjleonard37) [inline Examples/ObjectiveC/POIAlongRouteExample.m:112]: Same comment as for swift

File: Examples/Swift/LineStyleLayerExample.swift [hunk 4]
```diff
@@ -13,8 +13,9 @@ class LineStyleLayerExample_Swift: UIViewController, MGLMapViewDelegate {
 
         mapView.setCenter(
             CLLocationCoordinate2D(latitude: 45.5076, longitude: -122.6736),
-            zoomLevel: 11,
```
Comment (pjleonard37) [inline Examples/Swift/LineStyleLayerExample.swift:16]: Is this change needed for your example?

**Review by pjleonard37 — COMMENTED** [review]
@tsuz - this looks good to me. I added in a few questions, but overall I like the example. It worked well for me locally.

File: Examples/Swift/POIAlongRouteExample.swift (hunk 2 above)
Comment (tsuz) [reply Examples/Swift/POIAlongRouteExample.swift:123]: @pjleonard37 Good point. I updated this!

File: Examples/ObjectiveC/POIAlongRouteExample.m (hunk 3 above)
Comment (tsuz) [reply Examples/ObjectiveC/POIAlongRouteExample.m:112]: Also updated.

File: Examples/Swift/LineStyleLayerExample.swift (hunk 4 above)
Comment (tsuz) [reply Examples/Swift/LineStyleLayerExample.swift]: I removed this as it's not needed. Thanks for catching.

**Comment by tsuz** [comment]
@pjleonard37 @ZiZasaurus May I get your review on this?

File: Examples/ObjectiveC/POIAlongRouteExample.m [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    dashedLayer.lineCap = layer.lineCap;
+    dashedLayer.lineWidth = layer.lineWidth;
+    dashedLayer.lineColor = [NSExpression expressionForConstantValue:[UIColor whiteColor]];
+    dashedLayer.lineOpacity = [NSExpression expressionForConstantValue:@0.5];
+    // Dash pattern in the format [dash, gap, dash, gap, ...]. You’ll want to adjust these values based on the line cap style.
+    dashedLayer.lineDashPattern = [NSExpression expressionForConstantValue:@[@0, @1.5]];
+    
+    MGLSymbolStyleLayer *poiLayer = (MGLSymbolStyleLayer *)[self.mapView.style layerWithIdentifier:@"poi-label"];
+    [self.mapView.style insertLayer:layer belowLayer:poiLayer];
+    [self.mapView.style insertLayer:dashedLayer belowLayer:layer];
```
Comment (pjleonard37) [inline Examples/ObjectiveC/POIAlongRouteExample.m:108]: This is currently adding the dashed layer below the main layer so it is not visible. I'd recommend switching it to `aboveLayer` so it is visible and matches the Swift example.

**Review by pjleonard37 — APPROVED** [review]
Overall looks good after the one edit I commented on!
